public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts

```diff
@@ -108,15 +108,57 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
     return transformMetadataToKustoSchema(metadata, resourceUri);
   }
 
-  applyTemplateVariables(target: AzureMonitorQuery, scopedVars: ScopedVars): AzureMonitorQuery {
+  applyTemplateVariables(
+    target: AzureMonitorQuery,
+    scopedVars: ScopedVars,
+    testing?: boolean,
+    testResource?: string
+  ): AzureMonitorQuery {
     const item = target.azureLogAnalytics;
     if (!item) {
       return target;
     }
 
     const templateSrv = getTemplateSrv();
-    const resource = templateSrv.replace(item.resource, scopedVars);
+    // casting scopedVars to any for testing purposes.
+    const allVars = testing ? (scopedVars as any) : templateSrv.getVariables();
+    let resource = testing ? testResource : templateSrv.replace(item.resource, scopedVars);
     let workspace = templateSrv.replace(item.workspace, scopedVars);
+    let resPath = '';
+
+    // Complete the path for the selected resource if necessary
+    // using the type any below because not all properties available during runtime are declared
+    // as avilable properties in the VariableModel object
+    const resourceVar = allVars.find((var_: any) => var_.id === item.resource?.substring(1));
+    const resourceVarRawQuery = (resourceVar as any)?.query?.grafanaTemplateVariableFn.rawQuery;
+    var subscription = (allVars.find((var_: any) =>
+      var_.query.grafanaTemplateVariableFn.rawQuery.includes('Subscriptions(')
+    ) as any)?.current.value;
+
+    if (resourceVarRawQuery?.includes('Subscriptions(')) {
+      resPath = '/subscriptions/' + resource;
+      resource = resPath;
```

Comment:
`resPath` doesn't seem necessary. We can simply do `resource = '/subscriptions/' + resource;`
Same with below.

Reply:
done